tests/cupy_tests/math_tests/test_sumprod.py

```diff
@@ -126,10 +132,19 @@ def test_sum_dtype(self, xp, src_dtype, dst_dtype):
         a = testing.shaped_arange((2, 3, 4), xp, src_dtype)
         return a.sum(dtype=dst_dtype)
 
+    @testing.for_all_dtypes_combination(names=['src_dtype', 'dst_dtype'])
+    @testing.numpy_cupy_allclose()
+    def test_sum_keepdims_and_dtype(self, xp, src_dtype, dst_dtype):
+        if not xp.can_cast(src_dtype, dst_dtype):
+            return xp.array([])  # skip
+        a = testing.shaped_arange((2, 3, 4), xp, src_dtype)
+        return a.sum(axis=1, dtype=dst_dtype, keepdims=True)
```

Comment:
Did you mean to do this?
```suggestion
        return a.sum(axis=(1, 2), dtype=dst_dtype, keepdims=True)
```

Reply:
I don't think so? There is a case immediately below this called `test_sum_keepdims_multiple_axes` that uses ``axis=(1, 2)``

Reply:
But this case won’t use CUB to do reduction.

Reply:
okay, perhaps I can change it to just set `axis=2` then.